Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/webapp/WebAppWrapper.java

```diff
@@ -132,4 +134,44 @@ private WebApp getWebApp() {
             throw new RuntimeException("Failed to get webapp instance");
         }
     }
+
+    @Override
+    public void deploy(DeployType type, File file) {
+        this.getWebApp().deploy(type, file);
+    }
+
+    @Override
+    public Completable deployAsync(DeployType type, File file) {
+        return this.getWebApp().deployAsync(type, file);
+    }
+
+    @Override
+    public void deploy(DeployType type, File file, DeployOptions deployOptions) {
+        this.getWebApp().deploy(type, file, deployOptions);
+    }
+
+    @Override
+    public Completable deployAsync(DeployType type, File file, DeployOptions deployOptions) {
+        return this.getWebApp().deployAsync(type, file, deployOptions);
+    }
+
+    @Override
+    public void deploy(DeployType type, InputStream file) {
```

Comment:
bad parameter file for type InputStream

Reply:
the design is owned by java sdk in its interface.